File: problems/java/1/Solution1.java
package problems.java.1

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numsToIdx = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int comlement = target - nums[i];
            if (numsToIdx.containsKey(comlement)) {
                return new int[] { numsToIdx.get(comlement), i };
            }
            numsToIdx.put(nums[i], i);
        }
        throw new IllegalStateException("Invalid input");
    }
}

File: problems/swift/1/Solution1.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict = [Int: Int]()
        
        for (index, num) in nums.enumerated() {
            let expected = target - num
            if dict[expected] != nil {
                return [index, (dict[expected] ?? 0)]
            }
            dict[num] = index
        }
        return []
    }
}
